decidim-admin/app/controllers/decidim/admin/participatory_process_user_roles_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+require_dependency "decidim/admin/application_controller"
+
+module Decidim
+  module Admin
+    # Controller that allows managing all the Admins.
+    #
+    class ParticipatoryProcessUserRolesController < ApplicationController
+      def create
+        authorize! :create, Decidim::ParticipatoryProcessUserRole
+        @form = ParticipatoryProcessUserRoleForm.from_params(params)
+
+        CreateParticipatoryProcessAdmin.call(@form, participatory_process) do
+          on(:ok) do
+            flash[:notice] = I18n.t("participatory_process_user_roles.create.success", scope: "decidim.admin")
+          end
+
+          on(:invalid) do
+            flash.now[:alert] = I18n.t("participatory_process_user_roles.create.error", scope: "decidim.admin")
+          end
+          redirect_to participatory_process_path(participatory_process)
```

Comment:
We already talked about this offline: Having to redirect to `participatory_process#show` is a really sub-optimal UX. We'll be adding a sidebar in participatory processes in the near future that will allow splitting all the concerns in different controllers/views.